src/popup/PopupLayer.tsx

```diff
@@ -3,32 +3,37 @@ import noop from 'lodash/noop';
 import PopupReply from '../components/Popups/PopupReply';
 import { CreateArg as HighlightCreateArg } from '../highlight/actions';
 import { CreateArg as RegionCreateArg } from '../region/actions';
-import { CreatorItem, CreatorStatus, isCreatorStagedHighlight, isCreatorStagedRegion } from '../store';
+import { CreatorItem, CreatorStatus, isCreatorStagedHighlight, isCreatorStagedRegion, Mode } from '../store';
 import { PopupReference } from '../components/Popups/Popper';
 import { Shape } from '../@types';
 import './PopupLayer.scss';
 
 type Props = {
     createHighlight?: (arg: HighlightCreateArg) => void;
     createRegion?: (arg: RegionCreateArg) => void;
-    isCreating: boolean;
     isPromoting: boolean;
     location: number;
     message: string;
+    mode: Mode;
     popupReference?: Shape;
     resetCreator: () => void;
     setMessage: (message: string) => void;
     staged?: CreatorItem | null;
     status: CreatorStatus;
 };
 
+const modeStagedMap: { [M in Mode]?: Function } = {
```

Comment:
Can we handle this in the container? Ideally, presentation components shouldn't directly use selectors.